### DISCUSSION TITLE
How to Detect Multiple Tabs in Browser Extensions and Show Notification with Custom Link?

### DISCUSSION THREAD

**Discussion by usmanoptimize** [community/community#147081] [Programming Help]
### Select Topic Area

Question

### Body

I am developing a browser extension that aims to detect if the user opens multiple tabs of the same extension and show a notification with a custom link. Specifically, I'm working on an extension for the LESCO website ([https://lescobil.pk/](https://lescobil.pk/)) to help users check their electricity bills.

**Current Implementation:**
The extension is built to monitor tab openings and trigger a notification when the user opens more than one tab of the same extension. Here’s the code I have tried so far:
```
// Background script for Chrome extension
chrome.tabs.onCreated.addListener(function(tab) {
    chrome.tabs.query({}, function(tabs) {
        let tabCount = 0;
        tabs.forEach(t => {
            if (t.url.includes('https://lescobil.pk')) {
                tabCount++;
            }
        });

        if (tabCount > 1) {
            chrome.notifications.create({
                type: 'basic',
                iconUrl: 'icon.png',
                title: 'Multiple Tabs Detected!',
                message: 'You have opened more than one tab of LESCO bill checker. Please close the extra tab.',
                buttons: [{
                    title: 'Go to Website',
                    iconUrl: 'icon.png'
                }]
            });
        }
    });
});
```

**Problem:**
I’m facing issues with handling edge cases, especially when tabs are opened in different browsers (Chrome vs. Firefox). I need to ensure:
1. The extension performs consistently across browsers.
2. Notifications only trigger once even if multiple tabs are opened within a short period.
3. The link in the notification should work and should open the website in the foreground when clicked.

**Things I’ve Tried:**
1. Checked for cross-browser compatibility by testing the code in Chrome and Firefox. However, performance varies between them.
2. Added event listeners for chrome.tabs.onCreated but still encountering issues with multiple tabs being detected rapidly.

**Questions:**
1. Is there a more efficient way to detect multiple tabs of the same extension across different browsers (Chrome/Firefox)?
2. How can I prevent repeated notifications from being shown if a user opens multiple tabs in quick succession?
3. Are there any best practices for handling tab creation events in browser extensions with multiple tabs?

**Additional Information:**
Platform: Chrome, Firefox
Extension Purpose: Display notifications when multiple tabs of the LESCO bill checker extension are open simultaneously.
Link: The notification needs to include a link to [https://lescobil.pk/](https://lescobil.pk/).

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @usmanoptimize!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!
